mmv1/api/product.go

```diff
@@ -231,3 +232,93 @@ func (p *Product) TerraformName() string {
 func (p Product) Lineage() string {
 	return p.Name
 }
+
+func Merge(self, otherObj reflect.Value) {
```

Comment:
See the equivalent ruby merge functions here: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/api/object.rb#L32